File: Años bisiestos.java
import java.util.Scanner;

public class LeapYear {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("anno:");
        int year = input.nextInt();
        boolean isLeapYear;
        if (year < 1582) {
            isLeapYear = year % 4 == 0;
        } else {
            isLeapYear = year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
        }
        if (isLeapYear) {
            System.out.println(year + " es bisiesto");
        } else {
            System.out.println(year + " no es bisiesto");
        }
    }
}

File: Años bisiestos.py
year = int(input("anno:"))
if year < 1582:
    is_leap_year = year % 4 == 0
else:
    is_leap_year = year % 4 == 0 and (year % 100 != 0 or year % 400 == 0)
if is_leap_year:
    print(year, "es bisiesto")
else:
    print(year, "no es bisiesto")